src/contextmenu.js

```diff
@@ -0,0 +1,472 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Multiple selection context menu.
+ */
+
+import * as Blockly from 'blockly/core';
+import {blockSelection, hasSelectedParent} from './global';
+
+/**
+ * Modification for context menu 'Duplicate' to be available for
+ * multiple blocks.
+ */
+const registerDuplicate = function() {
+  const duplicateOption = {
+    displayText: function() {
+      const selectedBlocksLength = blockSelection.size;
+      if (selectedBlocksLength <= 1) {
+        return Blockly.Msg['DUPLICATE_BLOCK'];
+      } else {
+        return Blockly.Msg['DUPLICATE_BLOCK'] + ' (' +
+        selectedBlocksLength + ')';
+      }
+    },
+    preconditionFn: function(scope) {
+      const block = scope.block;
+      if (!block.isInFlyout && block.isDeletable() && block.isMovable()) {
+        if (block.isDuplicatable()) {
+          return 'enabled';
+        }
+        return 'disabled';
+      }
+      return 'hidden';
+    },
+    callback: function(scope) {
```

Comment:
Can you add a comment here explaining that only duplicate-able blocks will be duplicated?